### PR TITLE
[WIP] Feature: Extractor Catalog

### PR DESCRIPTION
**Pull request by bodom0015**
## Description
Extractor Catalog Design - Catalog View

Screen Shot 2020-08-31 at 7 54 57 PM

### Problem
We currently have a very primitive table view for our Extractor Catalog. We plan on evolving it into a more capable view, but there may be merge conflicts if too many things happen here at once.

### Approach
The purpose of this PR is to (hopefully) minimize conflict down the road by creating a basic layout and stubbing out any missing logic.

This task is blocking near-future development towards the Extractor Catalog.

### How to Test
Prerequisites: 2 users (1 admin + 1 non-admin), at least one extractor registered

1. Checkout and run this branch locally
2. Sign-up a non-admin user (if you haven't already) and login to Clowder as non-admin
3. Enable the RabbitMQ plugin and register an extractor, if you haven't already
4. From the navbar, choose "Explore > Extractors"
    * You should be brought to a read-only version of the Extractor Catalog
    * You should see each extractor's Name, Authors, Version, Maturity (stubbed), My/All Job Counts (stubbed), and Ratings (stubbed)
    * You should see that a dropdown menu is offered for each extractor
    * You should see a Labels and Filters section
    * You should be able to see (but not modify) which extractors are currently enabled on Clowder
    * You should **not** see a "Manage" button at the top-right of the Labels section
    * You should **not** see any buttons beneath the list of extractors
5. Expand the dropdown menu beside an extractor
    * You should see a dropdown menu appear
    * You should see the dropdown offers a working option to "View Details"
    * You should see the dropdown offers (stubbed) options to "View History", "View Logs", "View Metrics", and "Rate & Comment"
    * You should **not** see (stubbed) options to "Move to Development", "Move to Staging", or "Move to Production" (these are admin-only)
6. Logout and log back into Clowder as an admin
7. From the navbar, choose "Explore > Extractors"
    * You should be brought to the writable version of the enable/disable extractors list which preserves existing admin functionality
    * You should be able to check/uncheck which extractors are currently enabled on Clowder
    * You should see each extractor's Name, Authors, Version, Maturity (stubbed), My/All Job Counts (stubbed), and Ratings (stubbed)
    * You should see that a dropdown menu is offered for each extractor
    * You should see a Labels and Filters section, with a "Manage" button at the top-right of the Labels section (admin-only)
    * You should see Save/Cancel buttons beneath the list of extractors
8. Expand the dropdown menu beside an extractor
    * You should see a dropdown menu appear
    * You should see the dropdown offers a working option to "View Details"
    * You should see the dropdown offers (stubbed) options to "View History", "View Logs", "View Metrics", and "Rate & Comment"
    * You should see (stubbed) options to "Move to Development", "Move to Staging", or "Move to Production" (these are admin-only)

## Review Time Estimate

- [x] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have signed the CLA
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by lmarini** [comment]
Should we mark this as draft? A lot of the functionality is not implemented yet?

**Review by lmarini — CHANGES_REQUESTED** [review]
A couple of quick comments:

- the rounded corners look a bit faded in chrome. since we don't use them anywhere else we could remove them OR you can try increasing the size and see if they look better
- what will the manage link be used for? It looks funny in that location, but maybe it's because it's the first time we a link on a border like that?

**Comment by bodom0015** [comment]
@lmarini this PR should  not be a draft, and should be merged. It lays the groundwork necessary for future sprints, and was explicitly requested by @robkooper. We can hide the stubbed sections for now using the `showOptional` parameters.

Regarding your other feedback:
* Did you mean the fieldset/legend border around the Filters and Labels sections? If so, there is a slight shadow on them that was an accident, but I ended up liking the way it looked. If you disagree, I could try to restyle them to drop the `box-shadow` CSS? Or have I misunderstood which "rounded corners" you meant?
* Per the mockups, this button is just a link off to the view to create/assign these labels. This button did appear in the mockups.. I could move it above the Labels section, if that might be less pattern-breaking?

File: public/stylesheets/extractions.css [hunk 1]
```diff
@@ -30,4 +30,29 @@
...
+    box-shadow:  0px 0px 0px 0px #000;
+}
+
+legend.labels-border, legend.filters-border {
+    font-size: 1.2em !important;
+    text-align: left !important;
+    width:inherit; /* Or auto */
+    padding:0 10px; /* To give a bit of padding on the left and right */
+    border-bottom:none;
+    border-radius: 10px;
```
Comment (bodom0015) [inline public/stylesheets/extractions.css:52]:
```suggestion
    border-radius: 4px;
```

**Comment by lmarini** [comment]
@bodom0015 @max-zilla @bingzhang all outstanding PRs have been merged in this one. Can you give it one more review and comment out the uninplemented features?

File: CONTRIBUTORS.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
 Following is a list of contributors in alphabetical order:
 
-- Aaraj Habib
 - Ashwini Vaidya
```
Comment (max-zilla) [inline CONTRIBUTORS.md:3]: is this removal intentional?

**Comment by max-zilla** [comment]
I ran the branch and the catalog itself looks fine, but in all the merges it looks like CHANGELOG and our contributors file got a bit mangled, I don't see 1.11 on it.

File: CONTRIBUTORS.md (hunk 2 above)
Comment (bodom0015) [reply CONTRIBUTORS.md:3]: This was definitely not intentional. Perhaps I botched a merge fix somewhere? Thank you, I will add this person back to the list

File: app/services/GraylogService.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
...d\": \"7c14f490-f60c-11ea-9775-0242ac120006\",\n      \"facility\": \"user-level\",\n      \"timestamp\": \"2020-09-13T22:00:06.000Z\"\n    },\n    \"index\": \"graylog_450\",\n    \"decoration_stats\": null\n  }, {\n    \"highlight_ranges\": {},\n    \"message\": {\n      \"process_id\": \"5822\",\n      \"gl2_accounted_message_size\": 338,\n      \"level\": 3,\n      \"gl2_remote_ip\": \"10.0.10.241\",\n      \"gl2_remote_port\": 33366,\n      \"streams\": [\"000000000000000000000001\"],\n      \"gl2_message_id\": \"01EJ4GV93STR68E7FRCRM7Q1A5\",\n      \"source\": \"b6ae36cb3653\",\n      \"message\": \"2020-09-13 20:00:27,105 [Thread-68      ] INFO    : pyclowder.connectors - [5f5e7a458f08aa01dd476172] : StatusMessage.start: Started processing\",\n      \"gl2_source_input\": \"5f2d6cbdfe6eb70769683d1c\",\n      \"application_name\": \"clowder-dev_ncsa_file_digest.2.s6i35ypkuyzau2pr8\",\n      \"gl2_source_node\": \"f1971377-bf1a-4d91-a30b-f784a03946bb\",\n      \"_id\": \"b869f690-f5fb-11ea-9775-0242ac120006\",\n      \"facility\": \"user-level\",\n      \"timestamp\": \"2020-09-13T20:00:27.000Z\"\n    },\n    \"index\": \"graylog_450\",\n    \"decoration_stats\": null\n  }]\n}"
```
Comment (lmarini) [inline app/services/GraylogService.scala:49]: @bingzhang what's this for?

Comment (bingzhang) [reply app/services/GraylogService.scala:49]: This is my debug code. I will make another pull request to remove them.

**Review by lmarini — APPROVED** [review]
I was unable to test the graylog functionality on my machine. Everything else looks good. We will merge and test the graylog functionality on the dev sites where it is already setup.
